src/Plugin/pathauto/AliasType/FileAliasType.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\pathauto\Plugin\AliasType\NodeAliasType.
+ */
+
+namespace Drupal\file_entity\Plugin\pathauto\AliasType;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\file\Entity\File;
+use Drupal\pathauto\AliasTypeBatchUpdateInterface;
+use Drupal\pathauto\Plugin\pathauto\AliasType\EntityAliasTypeBase;
+
+/**
+ * A pathauto alias type plugin for content entities.
+ *
+ * @AliasType(
+ *   id = "file",
+ *   label = @Translation("File"),
+ *   types = {"file"},
+ *   provider = "file",
```

Comment:
the provider is not necessary I think, it defaults to the module that contains the plugin.